File: front-end/src/app/entities/don.ts
import { ReceveurDon } from './receveur-don';
import { Adherent } from "./adherent";
import { Donneur } from "./donneur";

export class Don {
  constructor(private _id_don?:Number,
              private _date_don?:Date,
              private _description_don?:string,
              private _type_don?:string,
              private _date_sortie?:Date,
              private _lieu_sortie?:string,
              private _montant?:Number,
              private _num_compte_banque?:Number,
              private _date_cheque?:Date,
              private _nom_banque?:string,
              private _etat?:string,
              private _suivi?:boolean,
              private _nom_complet?:string,
              private _role?:string,
              private _receveur_don?:ReceveurDon,
              private _donneur?:Donneur
  ){
   this._donneur=new Donneur();
   this._receveur_don=new ReceveurDon();
  }

  get id_don(){
    return this._id_don;
  }
  set id_don(value:Number){
    this._id_don=value;
  }
  get date_don(){
    return this._date_don;
  }
  set date_don(value:Date){
    this._date_don=value;
  }
  get description_don(){
    return this._description_don;
  }
  set description_don(value:string){
    this._description_don=value;
  }
  get type_don(){
    return this._type_don;
  }
  set type_don(value:string){
    this._type_don=value;
  }
  get dateSortie(){
    return this._date_sortie;
  }
  set dateSortie(value:Date){
    this._date_sortie=value;
  }
  get lieuSortie(){
    return this._lieu_sortie;
  }
  set lieuSortie(value:string){
   this._lieu_sortie=value;
  }
  get montant(){
    return this._montant;
  }
  set montant(value:Number){
    this._montant=value;
  }
  get num_compte_banque(){
    return this._num_compte_banque;
  }
  set num_compte_banque(value:Number){
    this._num_compte_banque=value;
  }
  get date_cheque(){
    return this._date_cheque;
  }
  set date_cheque(value:Date){
    this._date_cheque=value;
  }
  get nom_banque(){
    return this._nom_banque;
  }
  set nom_banque(value:string){
    this._nom_banque=value;
  }
  get etat(){
    return this._etat;
  }
  set etat(value:string){
    this._etat=value;
  }
  get suivi():boolean{
    return this._suivi;
  }
  set suivi(value:boolean){
    this._suivi=value;
  }
  get receveur_don(){
    return this._receveur_don;
  }
  set receveur_don(value:ReceveurDon){
    this._receveur_don=value;
  }
  get donneur(){
    return this._donneur;
  }
  set donneur(value:Donneur){
    this._donneur=value;
  }
  get nomComplet(){
    return this._nom_complet;
  }
  set nomComplet(value:string){
    this._nom_complet=value;
  }
  get role(){
    return this._role;
  }
  set role(value:string){
    this._role=value;
  }
}

File: back-end/src/main/java/ma/infosat/gestion/adherent/cotisation/entities/Don.java
package ma.infosat.gestion.adherent.cotisation.entities;

import java.io.Serializable;
import java.time.LocalDate;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.DiscriminatorColumn;
import javax.persistence.DiscriminatorType;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Inheritance;
import javax.persistence.InheritanceType;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;

import com.fasterxml.jackson.annotation.JsonSubTypes;
import com.fasterxml.jackson.annotation.JsonSubTypes.Type;
import com.fasterxml.jackson.annotation.JsonTypeInfo;
@Entity
@Inheritance(strategy=InheritanceType.SINGLE_TABLE)
@DiscriminatorColumn(name="type_don",discriminatorType=DiscriminatorType.STRING,length=4)
@JsonTypeInfo(use=JsonTypeInfo.Id.NAME,include=JsonTypeInfo.As.PROPERTY,property="type_don")
@JsonSubTypes({
	@Type(name="matr",value=Materiel.class),
	@Type(name="cheq",value=Cheque.class),
	@Type(name="espc",value=Espece.class),
	@Type(name="vers",value=Versement.class)
})
public abstract class Don implements Serializable {
	/**
	 * 
	 */
	private static final long serialVersionUID = 1L;
	
	
	@Id
	@GeneratedValue(strategy=GenerationType.IDENTITY)
	
	protected int id_don;
	protected LocalDate date_don;
	@Column(nullable=true,updatable=true)
	protected double montant;
 	protected String description_don;
	@ManyToOne(cascade=CascadeType.MERGE)
	@JoinColumn(name="id_donneur")
	protected Donneur donneur;
	@ManyToOne(cascade=CascadeType.MERGE)
	@JoinColumn(name="id_receveur_don")
	protected ReceveurDon receveur_don;
	
	public Don() { }
	
	public Don(LocalDate date_don, String description_don) {
		super();
		this.date_don = date_don;
		this.description_don = description_don;
	}
	

	public Don(LocalDate date_don, String description_don, Donneur donneur, ReceveurDon receveur_don) {
		super();
		this.date_don = date_don;
		this.description_don = description_don;
		this.donneur = donneur;
		this.receveur_don = receveur_don;
	}

	public ReceveurDon getReceveur_don() {
		return receveur_don;
	}

	public void setReceveur_don(ReceveurDon receveur_don) {
		this.receveur_don = receveur_don;
	}

	public LocalDate getDate_don() {
		return date_don;
	}
	public void setDate_don(LocalDate date_don) {
		this.date_don = date_don;
	}
	public String getDescription_don() {
		return description_don;
	}
	public void setDescription_don(String description_don) {
		this.description_don = description_don;
	}
	public int getId_don() {
		return id_don;
	}

	public Donneur getDonneur() {
		return donneur;
	}

	public void setDonneur(Donneur donneur) {
		this.donneur = donneur;
	}

	public double getMontant() {
		return montant;
	}

	public void setMontant(double montant) {
		this.montant = montant;
	}

	public void setId_don(int id_don) {
		this.id_don = id_don;
	}
	

}
